src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -170,6 +171,7 @@ public int GetHashCode(object obj)
     }
 
     [Serializable]
+    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     internal sealed class CultureAwareComparer : StringComparer
```

Comment:
Here are the internal types listed. Without them we can't serialize Comparers, EqualityComparers, not even Exception.

```
CultureAwareComparer 
OrdinalComparer
OrdinalIgnoreCaseComparer
GenericComparer<T>
NullableComparer<T>
ObjectComparer<T>
Int32EnumComparer<T>
UInt32EnumComparer<T>
Int64EnumComparer<T>
UInt64EnumComparer<T>
GenericEqualityComparer<T>
NullableEqualityComparer<T>
ObjectEqualityComparer<T>
NonRandomizedStringEqualityComparer
ByteEqualityComparer 
EnumEqualityComparer<T>
SByteEnumEqualityComparer<T>
ShortEnumEqualityComparer<T>
LongEnumEqualityComparer<T>
ListDictionaryInternal 
DictionaryNode 
__RestrictedErrorObject <-- Used for exception data
```